src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -81,7 +84,7 @@ public static IEnumerable<VisualStudioInstance> QueryVisualStudioInstances()
             IEnumerable<VisualStudioInstance> instances,
             VisualStudioInstanceQueryOptions options)
         {
-            return instances.Where(i => options.DiscoveryTypes.HasFlag(i.DiscoveryType));
+            return instances.Where(i => i != null && options.DiscoveryTypes.HasFlag(i.DiscoveryType));
```

Comment:
When is i null?

Reply:
`GetInstance` can return `null` and consequently `GetInstances` can enumerate `null`. I, too, find it confusing. It would make more sense to not enumerate missing/broken instances in the first place.

Reply:
null check has been moved to GetInstances